packages/headless/src/features/configuration/configuration-slice.ts

```diff
@@ -176,18 +90,170 @@ function handleUpdateBasicConfiguration(
+      matchCoveoOrganizationEndpointUrlAnyOrganization(
+        state.commerce.apiBaseUrl
+      )
+    ) {
+      state.commerce.apiBaseUrl = getDefaultCommerceEndpointBaseUrl(
+        organizationId,
+        environment
+      );
+    }
+  }
+}
+
+function handleUpdateSearchConfiguration(
+  state: ConfigurationState,
+  payload: UpdateSearchConfigurationActionCreatorPayload
+) {
+  if (!isNullOrUndefined(payload.proxyBaseUrl)) {
+    state.search.apiBaseUrl = payload.proxyBaseUrl;
+  }
+  if (!isNullOrUndefined(payload.locale)) {
+    state.search.locale = payload.locale;
+  }
+  if (!isNullOrUndefined(payload.timezone)) {
+    state.search.timezone = payload.timezone;
+  }
+  if (!isNullOrUndefined(payload.authenticationProviders)) {
+    state.search.authenticationProviders = payload.authenticationProviders;
+  }
+}
+
+function handleUpdateCommerceProxyBaseUrl(
+  state: ConfigurationState,
+  payload: UpdateProxyBaseUrlPayload
+) {
+  if (!isNullOrUndefined(payload.proxyBaseUrl)) {
+    state.commerce.apiBaseUrl = payload.proxyBaseUrl;
+  }
+}
+
+function handleUpdateAnalyticsConfiguration(
+  state: ConfigurationState,
+  payload: UpdateAnalyticsConfigurationActionCreatorPayload
+) {
+  if (!isNullOrUndefined(payload.enabled)) {
+    state.analytics.enabled = payload.enabled;
+  }
+  if (!isNullOrUndefined(payload.originContext)) {
+    state.analytics.originContext = payload.originContext;
+  }
+  if (!isNullOrUndefined(payload.originLevel2)) {
+    state.analytics.originLevel2 = payload.originLevel2;
+  }
+  if (!isNullOrUndefined(payload.originLevel3)) {
+    state.analytics.originLevel3 = payload.originLevel3;
+  }
+  if (!isNullOrUndefined(payload.proxyBaseUrl)) {
+    state.analytics.apiBaseUrl = payload.proxyBaseUrl;
+    state.analytics.nextApiBaseUrl = payload.proxyBaseUrl;
+  }
+  if (!isNullOrUndefined(payload.trackingId)) {
+    state.analytics.trackingId = payload.trackingId;
+  }
+  if (!isNullOrUndefined(payload.analyticsMode)) {
+    state.analytics.analyticsMode = payload.analyticsMode;
+  }
+  if (!isNullOrUndefined(payload.source)) {
+    state.analytics.source = payload.source;
+  }
+  const magicCookie = getMagicCookie();
+  if (magicCookie) {
+    state.analytics.analyticsMode = 'next';
+    state.analytics.trackingId = magicCookie;
+  }
+  if (!isNullOrUndefined(payload.runtimeEnvironment)) {
+    state.analytics.runtimeEnvironment = payload.runtimeEnvironment;
+  }
+  if (!isNullOrUndefined(payload.anonymous)) {
+    state.analytics.anonymous = payload.anonymous;
+  }
+  if (!isNullOrUndefined(payload.deviceId)) {
+    state.analytics.deviceId = payload.deviceId;
+  }
+  if (!isNullOrUndefined(payload.userDisplayName)) {
+    state.analytics.userDisplayName = payload.userDisplayName;
+  }
+  if (!isNullOrUndefined(payload.documentLocation)) {
+    state.analytics.documentLocation = payload.documentLocation;
+  }
+}
+
+function handleUpdateCommerceAnalyticsConfiguration(
```

Comment:
Why is this here?

And also why is this a duplicated sub-set of the `handleUpdateAnalyticsConfiguration` method above?

All of the stuff in the `UpdateCommerceAnalyticsConfigurationPayload` are also present in the `UpdateAnalyticsConfigurationActionCreatorPayload` why isn't one a sub-type of the other that would also be accepted in the `handleUpdateAnalyticsConfiguration` method?

Reply:
> Why is this here?

Because we didn't add a commerce-specific configuration slice

> why isn't one a sub-type of the other that would also be accepted in the handleUpdateAnalyticsConfiguration method?

Good idea! 👍